docs/installation/requirements.md

```diff
@@ -0,0 +1,74 @@
+- If automating real devices, additional manual configuration is required. Please check the
+  [Real Device Configuration](../preparation/real-device-config.md) document for more details.
+- Webviews must be debuggable in order to test them. If it is not possible to connect to your
+  webview(s) using [Safari remote debugger](https://appletoolbox.com/use-web-inspector-debug-mobile-safari/),
+  then the driver will not be able to identify them.
+
+## Optional Dependencies
+
+- [`xcpretty`](https://github.com/supermarin/xcpretty) can be used to make Xcode output easier to
+  read. It can be installed by running `gem install xcpretty`.
+- [`ffmpeg`](https://ffmpeg.org/) is used for test video recording. It can be installed using
+  [`brew`](https://brew.sh/): `brew install ffmpeg`
+- [`idb`](https://github.com/facebook/idb), [`go-ios`](https://github.com/danielpaulus/go-ios) and
+  [`tidevice`](https://github.com/alibaba/taobao-iphone-device) can be used to improve device interactions
+- [WIX AppleSimulatorUtils](https://github.com/wix/AppleSimulatorUtils) can be used to improve some
+  Simulator interactions
+- [`py-ios-device`](https://github.com/YueChen-C/py-ios-device) is required in several `mobile:`
+  extensions, and can improve the general testing experience for real devices
+
+## Validate Dependencies Using Doctor
+
+Since driver version 5.13.0, you can automate the validation for the most of the above requirements
+as well as various optional ones needed by driver extensions by running the
+`appium driver doctor xcuitest` command.
+
+## Xcode/iOS Version Support
+
+Apple regularly updates the XCTest framework (which is used by the XCUITest driver) and publishes
+the changes with Xcode and iOS. Many major and even some minor iOS versions include breaking changes
+in the XCTest API. The following table specifies the minimum required XCUITest driver version that
+is compatible with such Xcode/iOS versions:
+
+| Xcode/iOS version | Minimum required XCUITest driver version |
+| --- | --- |
+| Xcode 13 / iOS 15 | 3.48.0 |
+| Xcode 14-beta.3 / iOS 16 Beta | 4.7.4 |
+| Xcode 14.3 / iOS 16.4 | 4.21.7 |
+| Xcode 15 / iOS 17 | 4.35.0 |
+
+Eventually, in order to simplify the client code and to have access to newer platform features, the
+Appium team drops supports for legacy XCTest implementations. The following table lists Xcode/iOS
+versions that are no longer supported by the XCUITest driver, along with the most recent driver
+version that _does_ still supports them. The general aim is to support at least the _current major_
+platform version and the _previous major_ one.
+
+| Xcode/iOS version | Last compatible XCUITest driver version |
+| --- | --- |
+| Xcode 8 / iOS 10 | 2.95.0 |
+| Xcode 9 / iOS 11 | 2.133.1 |
+| Xcode 10-10.1 / iOS 12-12.1 | 3.31.1 |
+| Xcode 10.2 / iOS 12.2 | 3.56.3 |
+| Xcode 11 / iOS 13 | 4.3.2 |
+| Xcode 12 / iOS 14 | 4.27.2 |
```

Comment:
Actually I'm having second thoughts about adding iOS here. I mean, it's still possible to use a newer Xcode with an older iOS version (to a certain extent), and WDA deprecates only the older Xcode compatibility. Are there any commits where compatibility with older iOS versions has been explicitly dropped? Or is driver compatibility entirely based on Xcode, in which case the deprecated iOS versions can be taken from Xcode release notes? (e.g. Xcode 12 drops support for iOS 8)

Reply:
We can drop older version explicitly by updating xcode project (by specifying minimal compatible OS version), but we haven't so for now. (I don't have strong opinions to explicitly set since)
https://github.com/appium/WebDriverAgent/pull/534 is not the iOS version itself but an example of dropping older Xcode version explicitly. Then for example older Xcode cannot open that project.

![Screenshot 2024-01-12 at 10 18 12 AM](https://github.com/appium/appium-xcuitest-driver/assets/5511591/b8be54fa-77ff-4a5c-93e8-142477ed5107)

We dropped workarounds for older iOS version like https://github.com/appium/WebDriverAgent/compare/v4.15.1...v5.0.0 (https://github.com/appium/WebDriverAgent/blob/master/CHANGELOG.md#500-2023-05-14). Usually Xcode includes latest two major OS versions as simulator, thus we've been basically following that. For example in Xcode 14, then iOS 17 and 16. So latest two major Xcode support (Xcode 14 and 13) would be iOS 15 - 17 right now. But basically we have no strong opinion to explicitly "drop" the mini version support by updating the project file. But we drop workarounds to avoid XCTest side's issue for example for older versions. The supported means we keep workaround as well.

Reply:
Maybe we could put "minimal tested iOS version" as a new column I guess instead of current "iOS version"

Reply:
Ah, perfect, the WDA 5.0 PR is exactly what I was looking for. It seems that it removed various workarounds for iOS versions even older than 10, so it's entirely possible that the last WDA 4.x was still supporting those. I'll see how I can reflect this in the docs.